core/jazz_is-slack-channel-available/index.js

```diff
@@ -0,0 +1,169 @@
+  var errorHandler = errorHandlerModule();
+  var config = configModule(event);
+  logger.init(event, context);
+  logger.info(event);
+
+  var resObj = {
+    'is_available': false
+  };
+
+  try {
+    genericInputValidation(event)
+      .then((res) => requestToChannels(config, resObj, res))
+      .then((res) => {
+        cb(null, responseObj(res, event.query));
+      })
+      .catch(error => {
+        if (error.result === 'inputError') {
+          return cb(JSON.stringify(errorHandler.throwInputValidationError(error.message)));
+        } else if (error.result === 'unauthorized') {
+          return cb(JSON.stringify(errorHandler.throwUnauthorizedError(error.message)));
+        } else {
+          cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+        }
+      });
+  } catch (e) {
+    cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+  }
+};
+
+function genericInputValidation(event) {
+  logger.info("inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.method) {
+      reject({
+        result: "inputError",
+        message: "method cannot be empty"
+      });
+    }
+
+    if (event.method !== 'GET') {
+      reject({
+        result: "inputError",
+        message: "Unsupported method/request"
+      });
+    }
+
+    if (event.method === 'GET' && (!event.query || Object.keys(event.query).indexOf('slack_channel') === -1)) {
+      reject({
+        result: "inputError",
+        message: "Missing input parameter slack_channel"
+      });
+    }
+
+    if (event.method === 'GET' && !event.query.slack_channel) {
+      reject({
+        result: "inputError",
+        message: "slack_channel value can not be empty/undefined"
+      });
+    }
+
+    if (!event.principalId) {
+      reject({
+        result: "unauthorized",
+        message: "Unauthorized."
+      });
+    }
+
+    resolve(event.query.slack_channel);
+  });
+}
+
+function requestToChannels(config, resObj, channel_name) {
+  logger.info("Inside requestToChannels:" + channel_name);
+  return new Promise((resolve, reject) => {
+    var slack_token = "token=" + config.slack_channel_token;
+    //Pull endpoints from config file
+    var public_channel_url = config.public_channel_endpoint + slack_token;
+    var priv_channel_url = config.priv_channel_endpoint + slack_token;
+    var urlList = [];
+    urlList.push(get_response(public_channel_url, channel_name));
+    urlList.push(get_response(priv_channel_url, channel_name));
+    Promise.all(urlList)
+      .then(res => {
+        if (res.indexOf('true') !== -1) {
+          resObj.is_available = true;
+        } else {
+          resObj.is_available = false;
+        }
+        resolve(resObj);
+      })
+      .catch(error => {
+        logger.error("error: " + JSON.stringify(error));
+        reject(error);
+      });
+  });
+}
+
+function get_response(channel_url, channel_name) {
+  'use strict';
```

Comment:
why is this here. if your runtime is set to 6.1 then you dont need use strict. even if you have to use this statement this should be at the top of the module.

Reply:
Removed 'use strict'.